File: app/helpers/slack.py
from requests import post
from ..config import *


def invite_to_slack(contractor, email_created):
    # Wait until e-mail has been created for contractor
    email_created.wait()

    # TODO: Take me out for production!!!
    # return {"text": "Sent slack invite", "status": "success"}

    slack_url = "https://slack.com/api/users.admin.invite"
    payload = {"token": SLACK_TOKEN, "email": contractor.get_email()}

    try:
        slack_response = post(slack_url, params=payload)
        ok = slack_response.json()["ok"]
        if ok:
            return {"text": "Sent slack invite", "status": "success"}
        else:
            return {"text": "Problem sending slack invite", "status": "failure"}
    except:
        return {"text": "Problem sending slack invite", "status": "failure"}

File: helper/slack.js
var request = require('request-promise').defaults({simple: false});
var config = require('config');
var auth = require('../helper/auth');
var Promise = require('bluebird');

var slack = exports;

slack.inviteToSlack = (contractor) => {

  var slackUrl = config.get('slack.baseUrl') + '/users.admin.invite';
  var token = config.get('slack.token');

  return request.get({
    url: slackUrl,
    json: true,
    qs: {
      token: token,
      email: contractor.getEmail()
    },
  }).then( (response) => {
    if ('ok' in response) {
      if (response.ok === true) {
        return {'text': 'Invited to Slack', 'status': 'success'};
      } else {
        return {'text': 'Already invited ' + contractor.getEmail(), 'status': 'failure'};
      }
    } else {
      return {'text': 'Problem inviting to slack', 'status': 'failure'};
    }
  }).catch( (err) => {
    return {'text': 'Problem inviting to slack', 'status': 'failure'}
  });
};